### PR TITLE
FEDX-8 Last tweaks for v2 rulesets

### PR DESCRIPTION
**Pull request by evanweible-wf**
- Update v2 rulesets to reflect internal discussions
- Update changelog and readme

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Review by maxwellpeterson-wf — APPROVED** [review]

File: lib/v2.recommended.yaml [hunk 1]
```diff
@@ -202,7 +201,6 @@ linter:
     - prefer_spread_collections
     - prefer_void_to_null
     - provide_deprecation_message
-    - sort_pub_dependencies
```
Comment (dustyholmes-wf) [inline lib/v2.recommended.yaml:205]: sort_pub_dependencies still exists as warning in this file.

File: lib/v2.recommended.yaml [hunk 2]
```diff
@@ -165,7 +165,6 @@ linter:
     - avoid_type_to_string
     - avoid_types_on_closure_parameters
     - avoid_unused_constructor_parameters
-    - cascade_invocations
```
Comment (dustyholmes-wf) [inline lib/v2.recommended.yaml:168]: cascade_invocations is still warning above.

File: lib/v2.recommended.yaml [hunk 3]
```diff
@@ -219,5 +217,4 @@ linter:
     - unnecessary_this
     - use_function_type_syntax_for_parameters
     - use_rethrow_when_possible
-    - use_string_buffers
```
Comment (dustyholmes-wf) [inline lib/v2.recommended.yaml:222]: use_string_buffers is still warning above.

**Review by dustyholmes-wf — APPROVED** [review]

**Review by maxwellpeterson-wf — APPROVED** [review]

**Comment by evanweible-wf** [comment]
QA +1
@Workiva/release-management-p

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**